### DISCUSSION TITLE
Error [ERR_MODULE_NOT_FOUND]

### DISCUSSION THREAD

**Discussion by matus123** [vercel/next.js#40313] [Help]
hi,

i am trying to use @typedorm library (https://github.com/typedorm/typedorm) with nextjs... but there is problem when i use this library in nextjs apis... i am getting runtime error 

```
error - Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/home/matus/repos/typedorm-nextjs-example/node_modules/@typedorm/core/esm/public-api' imported from /home/matus/repos/typedorm-nextjs-example/node_modules/@typedorm/core/esm/index.js
    at new NodeError (node:internal/errors:371:5)
    at finalizeResolution (node:internal/modules/esm/resolve:416:11)
    at moduleResolve (node:internal/modules/esm/resolve:932:10)
    at defaultResolve (node:internal/modules/esm/resolve:1044:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:422:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:222:40)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:76:40)
    at link (node:internal/modules/esm/module_job:75:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}
```

i tried tinkering with tsconfig or webpack configs, but without any success

when i tried to use `next-transpile-modules` it started to work but getting these kind of warnings
```
warn  - ./node_modules/@typedorm/common/esm/src/helpers/load-package.js
Critical dependency: the request of a dependency is an expression

./node_modules/@typedorm/core/esm/src/classes/connection/connection-metadata-builder.js
Critical dependency: the request of a dependency is an expression
```

minimal repo: https://github.com/matus123/nextjs-typedorm-example

**Comment by icyJoseph** [comment]
Might be worth sharing this with the https://github.com/typedorm/typedorm repository. I am not sure, but it looks like there's some file extension missing when looking for `public-api`, and I am not super clear on how to get that working. I think it should be trying to import `@typedorm/core/esm/public-api.js`, IIRC.

**Reply by matus123** [reply]
i have asked the same question in their repo aswell - https://github.com/typedorm/typedorm/discussions/281

**Reply by ShubhamTheReactGeek** [reply]
I'm facing the same issue, Did you find any solution for this ?
